File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/list/DebtsModule.java
package com.androidcollider.easyfin.debts.list;

import android.content.Context;

import com.androidcollider.easyfin.common.managers.format.date.DateFormatManager;
import com.androidcollider.easyfin.common.managers.format.number.NumberFormatManager;
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager;
import com.androidcollider.easyfin.common.repository.Repository;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class DebtsModule {

    @Provides
    DebtsMVP.Model provideDebtsMVPModel(Repository repository,
                                        DateFormatManager dateFormatManager,
                                        NumberFormatManager numberFormatManager,
                                        ResourcesManager resourcesManager,
                                        Context context) {
        return new DebtsModel(repository, dateFormatManager, numberFormatManager, resourcesManager, context);
    }

    @Provides
    DebtsMVP.Presenter provideDebtsMVPPresenter(DebtsMVP.Model model) {
        return new DebtsPresenter(model);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/list/DebtsModule.kt
package com.androidcollider.easyfin.debts.list

import android.content.Context
import com.androidcollider.easyfin.common.managers.format.date.DateFormatManager
import com.androidcollider.easyfin.common.managers.format.number.NumberFormatManager
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager
import com.androidcollider.easyfin.common.repository.Repository
import dagger.Module
import dagger.Provides

/**
 * @author Ihor Bilous
 */
@Module
class DebtsModule {
    @Provides
    fun provideDebtsMVPModel(
        repository: Repository?,
        dateFormatManager: DateFormatManager?,
        numberFormatManager: NumberFormatManager?,
        resourcesManager: ResourcesManager?,
        context: Context?
    ): DebtsMVP.Model {
        return DebtsModel(
            repository,
            dateFormatManager,
            numberFormatManager,
            resourcesManager,
            context
        )
    }

    @Provides
    fun provideDebtsMVPPresenter(model: DebtsMVP.Model?): DebtsMVP.Presenter {
        return DebtsPresenter(model)
    }
}